### PR TITLE
better Addon.update_status - will upgrade and downgrade as appropriate

### PR DESCRIPTION
**Pull request by eviljeff**
Its a more complete `update_status` but could do with some specific tests beyond the few I had to change. Also, it's of questionable utility unless we remove all the instances where we set the status directly.

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1043,33 +1043,35 @@ def get_default_icon_url(self, size):
         )
 
     @write
-    def update_status(self, ignore_version=None):
+    def update_status(self, ignore_version=None, force_update=False):
```
Comment (diox) [inline src/olympia/addons/models.py]: You never pass this argument. IMHO we should never force the status - if tests require it we should modify their versions so that it doesn't happen, or just keep using `addon.update(status=xxx)` for the special edge cases we want to test.

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -1043,33 +1043,35 @@ def get_default_icon_url(self, size):
...
         self.reload()
 
-        if (self.status in [amo.STATUS_NULL, amo.STATUS_DELETED] or
-                self.is_disabled or self.is_persona()):
+        if (self.status in [amo.STATUS_DISABLED, amo.STATUS_DELETED] or
+                self.disabled_by_user or self.is_persona()):
             self.update_version(ignore=ignore_version)
```
Comment (diox) [inline src/olympia/addons/models.py]: Since we're touching the method, might as well comment where needed. I feel this is one of the places that require one :)

**Comment by diox** [comment]
I love the idea in principle but yeah, I don't think it's very useful if we don't use it everywhere.
